frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -0,0 +1,330 @@
+  const newData = getFilteredRows(allFilters, data);
+  return {
+    data: newData,
+    unfilteredData: data,
+  };
+};
+
+const CustomNodeTable: React.FC<CustomNodeTableProps> = ({ data, loaded, ocsProps }) => {
+  const columns = getColumns();
+  const [nodes, setNodes] = React.useState([]);
+  const [sortBy, setSortBy] = React.useState<ISortBy>({ index: 0, direction: 'asc' });
+  const [error, setError] = React.useState('');
+  const [inProgress, setProgress] = React.useState(false);
+  const [storageClass, setStorageClass] = React.useState('');
+  const [selectedNodesCnt, setSelectedNodesCnt] = React.useState(0);
+
+  React.useEffect(() => {
+    const selectedNode = _.filter(nodes, 'selected').length;
+    setSelectedNodesCnt(selectedNode);
+  }, [nodes]);
+
+  React.useEffect(() => {
+    const formattedNodes = getRows(data);
+    // pre-selection of nodes
+    if (loaded && !nodes.length) {
+      getPreSelectedNodes(formattedNodes);
+      setNodes(formattedNodes);
+    }
+    // for getting nodes
+    else if (formattedNodes.length) {
+      nodes.forEach((row) => {
+        if (row.selected) {
+          const index = formattedNodes.findIndex((r) => r.id === row.id);
+          formattedNodes[index].selected = true;
+        }
+      });
+      setNodes(formattedNodes);
+    }
+  }, [data, loaded, nodes]);
+
+  // sorting by name
+  const onSort = (event, index, direction) => {
+    event.preventDefault();
+    const sortedNodes = nodes.sort((n1, n2) => (n1.id < n2.id ? -1 : n1.id > n2.id ? 1 : 0));
+    setNodes(direction === SortByDirection.asc ? sortedNodes : sortedNodes.reverse());
+    setSortBy({ index, direction });
+  };
+
+  const onSelect = (event, isSelected, index, node) => {
+    event.stopPropagation();
+    let formattedNodes;
+    if (index === -1) {
+      formattedNodes = nodes.map((oneRow) => {
+        oneRow.selected = isSelected;
+        return oneRow;
+      });
+    } else {
+      formattedNodes = [...nodes];
+      const nIndex = formattedNodes.findIndex((r) => r.id === node.id);
+      formattedNodes[nIndex].selected = isSelected;
+    }
+    setNodes(formattedNodes);
+  };
+
+  const handleStorageClass = (sc) => {
+    setStorageClass(_.get(sc, 'metadata.name'));
+  };
+
+  // labeling the selected nodes
+  const labelNodes = (selectedNode: NodeKind[]) => {
+    const labelPath = '/metadata/labels';
+    const labelData = selectedNode.map((node: NodeKind) => {
+      const labels = SelectorInput.arrayify(_.get(node, labelPath.split('/').slice(1)));
+      const lblVal = { ...SelectorInput.objectify(labels), ...labelObj };
+      const patch = [
+        {
+          op: labels.length ? 'replace' : 'add',
+          value: lblVal,
+          path: labelPath,
+        },
+      ];
+      return k8sPatch(NodeModel, node, patch);
+    });
+    return labelData;
+  };
+
+  // tainting the selected nodes
+  const taintNodes = (selectedNode: NodeKind[]) => {
+    const taintData = selectedNode
```

Comment:
This is not named well as it's returning promises, not data. You should have a return type on the function as well

Reply:
ack

Reply:
ack